godot-ffi/src/conv.rs

```diff
@@ -30,6 +30,41 @@ pub const fn bool_to_sys(value: bool) -> sys::GDExtensionBool {
     value as sys::GDExtensionBool
 }
 
+/// Convert a list into a pointer + length pair. Should be used together with [`ptr_list_from_sys`].
+///
+/// If `list_from_sys` is not called on this list then that will cause a memory leak.
+#[cfg(since_api = "4.3")]
+pub fn ptr_list_into_sys<T>(list: Vec<T>) -> (*const T, u32) {
+    let len: u32 = list
+        .len()
+        .try_into()
+        .expect("list must have length that fits in u32");
+    let ptr = Box::leak(list.into_boxed_slice()).as_mut_ptr();
+
+    (ptr.cast_const(), len)
+}
+
+/// Get a list back from a previous call to [`ptr_list_into_sys`].
+///
+/// # Safety
+/// - `ptr` must have been returned from a call to `list_into_sys`.
+/// - `ptr` must not have been used in a call to this function before.
+/// - `ptr` must not have been mutated since the call to `list_into_sys`.
+/// - `ptr` must not be accessed after calling this function.
```

Comment:
These 3 can be summarized as
- `ptr` must be passed to this function exactly once and not used in any other context.